include/aspect/particle/world.h

```diff
@@ -74,6 +76,26 @@ namespace aspect
         void initialize();
 
         /**
+         * Return an iterator to the first particle.
+         */
+        ParticleIterator<dim> begin() const;
```

Comment:
Can you introduce a typedef
```
  typedef ParticleIterator<dim> particle_iterator;
```
(or just `iterator` if you want)? This is how containers commonly do things.